DFReader.py

```diff
@@ -176,6 +176,9 @@ def set_mult_ids(self, mult_ids, mult_lookup):
                     self.units[i] = MULT_TO_PREFIX[unitmult]+self.units[i]
                 else:
                     self.units[i] = "%.4g %s" % (unitmult, self.units[i])
+                if self.units[i] in FORMAT_TO_STRUCT:
+                    (label, mul, t) = FORMAT_TO_STRUCT[self.units[i]]
+                    self.msg_mults[i] = mul
 
```

Comment:
I think "if self.units[i] in FORMAT_TO_STRUCT:" is doing what you want by a bit of chance, as 'c' is both prefix for 'centi' and also code for 'int16_t * 100'. Maybe something more explicit such as this for lines 175-181 (not tested!):
```suggestion
                # If no unit is specified, apply the multiplier to the value,
                # instead of prefixing the unit
                if self.units[i] == "":
                    self.msg_mults[i] = mult_lookup[mult_ids[i]]
                elseif unitmult in MULT_TO_PREFIX:
                    self.units[i] = MULT_TO_PREFIX[unitmult]+self.units[i]
                else:
                    self.units[i] = "%.4g %s" % (unitmult, self.units[i])
```
This would then work in cases which have any multiplier (not just *100) and no unit.